### PR TITLE
Refactoring some of the ODK release docs

### PR DESCRIPTION
**Pull request by matentzn**
This includes in particular a more detailed breakdown on how to deal with development snapshot, minor and major releases. This commit is not complete yet, so watch out for subsequent commits with additions.

### CODE REVIEW

**Comment by gouttegd** [comment]
Looks great. Among the things to add would be details about how the minor releases are managed (e.g., create a `BRANCH-1.X-MAINTENANCE` branch after a `v1.X` release has been made, and cherry-pick changes from the master branch to that branch, etc.).

Let me know if you want me to do it.

**Comment by matentzn** [comment]
> Let me know if you want me to do it.

Sounds great! Feel free to push to this branch here, I wont work on it until a bit later next week!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -132,24 +132,54 @@ In order to build and publish ODK, you need the following:
 
 ## General SOP for ODK release and publication
 
+- There are three types of releases: major, minor and development snapshot.
+  - Major versions include changes to the workflow system.
+  - Minor versions include changes to tools, such as ROBOT our python dependencies.
```
Comment (anitacaron) [inline CONTRIBUTING.md:137]:
```suggestion
  - Minor versions include changes to tools, such as ROBOT or Python dependencies.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -132,24 +132,54 @@ In order to build and publish ODK, you need the following:
...
+  - Development snapshots reflect the current state of the `main` (`master`).
+- They all have slightly different procedures which we will detail below.
+
+### Major releases
+
+Major releases contain changes to the workflow system of the ODK, e.g. changes to the `Makefile` and various supporting scripts (e.g. run.sh, update_repo.sh).
+They require users to update there repository light with `make update_repo`
```
Comment (anitacaron) [inline CONTRIBUTING.md:144]:
```suggestion
They require users to update their repository with `make update_repo`
```

Comment (matentzn) [reply CONTRIBUTING.md]: Not yet ready for review :D But thanks anyways!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -132,24 +132,92 @@ In order to build and publish ODK, you need the following:
...
+* Push all last-minute changes (CHANGELOG and version number updates) to the `BRANCH-1.X-MAINTENANCE` branch.
+* Build and publish the images from the tip of the `BRANCH-1.X-MAINTENANCE` branch (same procedure as above to build and publish a major release).
+* Create a GitHub release from the tip of the `BRANCH-1.X-MAINTENANCE` branch, with a `v1.X.Y` tag.
+* Resume backporting changes to the `BRANCH-1.X-MAINTENANCE` until the time comes for the next minor release.
+
+### Development snapshot releases
+
+Development snapshot releases reflect the current state of the `main` (`master`) branch. They are routine releases where we do not do a lot of testing, and anyone using these
+do so at their own risk.
+The ODK team is not accountable for pipelines that fail because of changes to the development snapshot.
```
Comment (gouttegd) [inline CONTRIBUTING.md:207]: I’d rather not call them “routine releases”. I guess it’s subject to interpretation, but to me, saying that something is done “routinely” kind of implies a certain form a stability, as in “it’s all routine stuff, there’s nothing surprising happening in those releases, and therefore nothing that could break your workflows” – exactly the opposite of what the `-dev` ”releases” are!

In fact I would not call them “releases” at all. They are “development snapshots”, not “development snapshot releases”.

My suggestion:

```
### Development snapshots

Development snapshots reflect the current state of the main (`master`) branch. They do not undergo the same level of testing (or any testing at all) as the normal releases, and are intended to help trialing and debugging the changes that happen in the `master` branch.

Development snapshots should _not_ be used in a production environment. Feel free to use them if you want to help us developing the next major release, but if you use them in your production pipelines, understand that you’re doing so at your own risk.

**Review by gouttegd — APPROVED** [review]
All good, except the bit below about calling the development snapshots “routine releases”.

Not blocking the merge for that, though – your call if you want to change it.

File: CONTRIBUTING.md (hunk 3 above)
Comment (matentzn) [reply CONTRIBUTING.md]: Makes sense. Done!

**Review by gouttegd — APPROVED** [review]

**Pull request merged**